### PR TITLE
postman stuff

### PR DESCRIPTION
**Pull request by HassanChowdhry**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import fetch from 'node-fetch';
+
+import xlsx from 'xlsx';
+
+const search = 'q=dogs&num=100';
+
+fetch(`https://google-search3.p.rapidapi.com/api/v1/search/${search}`, {
+  method: 'GET',
+  headers: {
+    // key
```
[the remark this answers is not in the document]
Comment (HassanChowdhry) [reply index.js:11]: make .env file

**Review by HassanChowdhry — COMMENTED** [review]
Remove GoogleLinks.xlsx file

once u merge the other PR, raise this one against MASTER/Main

File: .gitignore [hunk 2]
```diff
@@ -352,3 +352,5 @@ MigrationBackup/
 .idea
 
 .vscode
+
+.env
```
Comment (BilalIshtiaque) [inline .gitignore:356]: gud boi

File: index.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+function listToArrayOfArray(searchItems) {
+  const results = [['Title', 'Links']];
+  searchItems.forEach((searchItem) => {
+    results.push([searchItem.title, searchItem.link]);
+  });
+  return results;
+}
+
+function createExcel(arrayOfArray) {
+  const fileName = 'GoogleLinks.xlsx';
```
Comment (BilalIshtiaque) [inline index.js:80]: perhaps we can make the file name based on the search word used....name the file name based on what the user wants to search

File: index.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+class SearchItem {
+
+  constructor(title, description, link) {
+    this.title = title;
+    this.description = description;
+    this.link = link;
+  }
+
+}
```
Comment (BilalIshtiaque) [inline index.js:21]: move this to another file and import it here :)

**Review by BilalIshtiaque — CHANGES_REQUESTED** [review]
Boi

File: index.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    message: 'How many search results do you want? (max 100)', 
+    default() {
+      return 'Number';
+    },
+  });
+  searchNumber = answers.search_number;
+}
+
+await searchNameGetter();
+await searchNumberGetter();
```
[the remark this answers is not in the document]
Comment (HassanChowdhry) [reply index.js:51]: tried that.
couldnt figure it out tho, will try again a bit later.

File: index.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  name: 'search_name',
+  type: 'input',
+  message: 'What do you want to search in google?', 
+})).search_name;
+
+let searchNumber = (await inquirer.prompt({
+  name: 'search_number',
+  type: 'input',
+  message: 'How many search results do you want? (max 100)', 
+})).search_number;
```
Comment (BilalIshtiaque) [inline index.js:34]: ideally we want validation that the user added something along that range. write a TODO to do that later

File: index.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+//   headers: {
+//     'x-rapidapi-key': process.env.API_KEY,
+//     'x-proxy-location': 'US',
+//   },
+// })
+//   .then((res) => res.json())
+//   .then((res) => mapToSearchItems(res))
+//   .then((searchItems) => toExcelArrayOfArray(searchItems))
+//   .then((arrayOfArray) => createExcel(arrayOfArray))
+//   .catch((error) => console.log('error', error));
```
Comment (BilalIshtiaque) [inline index.js:49]: not being used ? not gonna use in future ? Into the trash bin it goes! delete

File: index.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      'x-rapidapi-key': process.env.API_KEY,
+      'x-proxy-location': 'US',
+    }, 
+  });
+  const jsonResponse = await response.json();
+  const searchItems = await mapToSearchItems(jsonResponse);
+  const arrayOfArray = await toExcelArrayOfArray(searchItems);
+  await createExcel(arrayOfArray);
+}
+await loadSearchResults();
```
Comment (BilalIshtiaque) [inline index.js:64]: dont see why we need to put all this in a method, why not jst leave it bare in script ? just write a comment before the block of code that you are searching the result n etc...

File: index.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+
+async function loadSearchResults() {
+  const response = await fetch(`https://google-search3.p.rapidapi.com/api/v1/search/${search}`, {
+    method: 'GET',
+    headers: {
+      'x-rapidapi-key': process.env.API_KEY,
+      'x-proxy-location': 'US',
+    }, 
+  });
+  const jsonResponse = await response.json();
```
Comment (BilalIshtiaque) [inline index.js:59]: u could combine this
(await fetch()).json()

**Review by BilalIshtiaque — APPROVED** [review]
Looks good, add your ReadMe file as instructed in slack.

File: ExcelUtils.js [hunk 10]
```diff
@@ -12,7 +12,7 @@ export function createExcel(arrayOfArray, searchName) {
   let name = searchName;
   name = name.split(' ').join('');
 
-  const fileName = `${name}.xlsx`;
+  const fileName = `output/${name}.xlsx`;
```
Comment (HassanChowdhry) [inline ExcelUtils.js:15]: what was the point of this?
It blocks the it blocks the creation of the excel file btw

**Pull request merged**